thavalon-webapp/src/components/gameComponents/gameHeader.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from "react";
+import { GameSocket, InboundMessage, InboundMessageType } from "../../utils/GameSocket";
+import { GameMessage, GameMessageType, Snapshot, NextProposalMessage } from "./constants";
+
+import "../../styles/gameStyles/gameGlobals.scss";
+
+/**
+ * An enum of game phase header titles.
+ */
+enum GamePhaseHeader {
+    ProposalOther = "Waiting for Proposal",
+    ProposalSelf = "Propose",
+    Vote = "Vote",
+    VoteResults = "Voting Results",
+    Mission = "Questing",
+    Assassination = "Assassination"
+}
+
+/**
+ * Creates the gamephase header and force count component.
+ */
+export function GameHeader(): JSX.Element {
+    // State for the current phase of the game
+    const [gamePhaseHeader, setGamePhaseHeader] = useState(GamePhaseHeader.ProposalOther);
+    // State for the number of turns until force. Initialized to 1 to prevent any force-specific actions from occurring.
+    const [turnsUntilForce, setTurnsUntilForce] = useState(1);
+    // State to determine the name of this player.
+    const [me, setMe] = useState("");
+
+    useEffect(() => {
+        const connection = GameSocket.getInstance();
+        connection.onLobbyEvent.subscribe(handleMessage);
+        connection.onGameEvent.subscribe(handleMessage);
+        return () => {
+            connection.onLobbyEvent.unsubscribe(handleMessage);
+            connection.onGameEvent.unsubscribe(handleMessage);
+        }
+    });
+
+    /**
+     * Handles any message received from the server.
+     * @param message An inbound message from the server
+     */
+    function handleMessage(message: InboundMessage): void {
+        switch (message.messageType) {
+            // If it's a snapshot, check the last message to get the current phase.
+            case InboundMessageType.Snapshot:
+                const snapshot = message.data as Snapshot;
+                setMe(snapshot.me);
+                const lastLogIndex = snapshot.log.length - 1;
+                const lastMessage = snapshot.log[lastLogIndex];
+                mapMessageToState(lastMessage);
+                // The last message will always give us the state, but to figure out
+                // the number of proposals remaining, we need a NextProposal message.
+                // Loop backwards over the messages until we find one.
+                if (lastMessage.messageType !== GameMessageType.NextProposal) {
+                    // We only hit this if the last messageType isn't a NextProposal,
+                    // so start the loop from the second to last element.
+                    for (let i = lastLogIndex - 1; i >= 0; i--) {
```

Comment:
if you start this at lastLogIndex instead of lastLogIndex - 1, can't you remove the if statement on line 56? Since this will loop from last to first message